Source/KSCrash/Recording/Tools/KSJSONCodec.c

```diff
@@ -1137,8 +1137,19 @@ int ksjsoncodec_i_decodeElement(const char** ptr,
                 return KSJSON_ERROR_INCOMPLETE;
             }
 
+            // our buffer is not necessarily NULL-terminated, so
+            // it would be undefined to call sscanf/sttod etc. directly.
+            // instead we create a temporary string.
             double value;
-            sscanf(start, "%lg", &value);
+            size_t len = (unsigned)(*ptr - start);
+            char * buf = malloc(len + 1);
```

Comment:
Out of interest, what causes json decoding to happen in async-safe code?

It'd be basically impossible to avoid the malloc in decoding a string in the general case (in the fastCopy case we could just return a pointer into the original JSON).